File: Arrays/1.TwoSumOpt.py

def twoSum(nums, target):
    hashMap = dict()
    for p1 in range(len(nums)):
        if nums[p1] in hashMap.keys():
            return [hashMap[nums[p1]], p1]
        numToFind = target - nums[p1]
        hashMap[numToFind] = p1
    return None


print(twoSum(nums=[1, 3, 7, 9, 2], target=11))

File: Arrays/1/1.TwoSumOpt.js

const findTwoSum = (nums, target) => {
  let hashMap = new Map()
  for(let p1=0;p1 < nums.length; p1++) {
    if (hashMap.has(nums[p1])) {
      return [hashMap.get(nums[p1]), p1]
    }
    let numToFind = target - nums[p1]
    hashMap.set(numToFind, p1)
  }
  return null;
}

console.log(findTwoSum((nums = [1, 3, 7, 9, 2]), (target = 11)));